README.md
## MBTagView


This library is derived from [SKTagView](https://github.com/zsk425/SKTagView).

### Installation with CocoaPods

```ruby
platform :ios, '7.0'
pod "MBTagView"
```

### Usage

#### Example
Check out the project. It contains the usages of normal way and in UITableViewCell.

#### Code
```objc
- (void)setupTagView
{
    self.tagView = ({
        MBTagView *view = [MBTagView new];
        view.backgroundColor = [UIColor colorWithHexString:@"#EAEAEA"];
        view.padding    = UIEdgeInsetsMake(10, 10, 10, 35);
        view.insets    = 10;
        view.lineSpace = 10;
        __weak MBTagView *weakView = view;
        weakView.didClickTagAtIndex = ^(NSUInteger index)  {

        };
        weakView.animateBlockForEachView = ^(UIView *v) {
            CGAffineTransform transform = v.transform;
            [UIView animateWithDuration:0.1f delay:0.0f options: UIViewAnimationOptionAutoreverse| UIViewAnimationOptionCurveLinear animations:^{
                v.transform = CGAffineTransformScale(transform, 0.95, 0.95);
            } completion:^(BOOL finished) {
                if (finished) {
                    v.transform = CGAffineTransformScale(transform, 1, 1);
                }
            }];
        };
        view;
    });
    [self.view addSubview:self.tagView];

    [self.tagView mas_makeConstraints:^(MASConstraintMaker *make) {
        UIView *superView = self.view;
        make.top.equalTo(superView).offset(40);
        make.leading.equalTo(superView.mas_leading);
        make.trailing.equalTo(superView.mas_trailing);
    }];

    [@[@"Sent", @"Attachments", @"Today", @"Conversations", @"Archieve",@"Trash", @"Others", @"Newsletters"] enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop)
     {
         MBTag *tag = [MBTag tagWithText:obj];
         tag.textColor = [UIColor colorWithHexString:@"#9B9B9B"];
         tag.bgColor = UIColor.whiteColor;
         tag.selectedbackgroundColor = [UIColor colorWithHexString:@"#83D5F6"];
         tag.selectable = YES;
         tag.cornerRadius = 13;
         tag.animable = YES;
         tag.fontSize = 15;
         tag.padding = UIEdgeInsetsMake(5, 12, 5, 12);

         [self.tagView addTag:tag];
     }];
}
```

### Tips
When uses with UITableViewCell in multi-line mode,it MUST be set preferredMaxLayoutWidth before invoking
```objc
[cell.contentView systemLayoutSizeFittingSize:UILayoutFittingCompressedSize].height + 1;
```

### Screenshots

![](https://raw.github.com/mailburn/MBTagView/master/Screenshots/example.png)

### License

MBTagView is available under the MIT license. See the LICENSE file for more info.


CHANGELOG.md
###v1.0
Added support for selected state color, animate of each tag when selecting.

###v0.7
Added support for attributed strings

###v0.6
Added support for setting the font.

Fixed bug when tag's title is too long.

###v0.5
Added support for setting the userInteractionEnabled of tag.

###v0.4
Added support for setting bgImg,borderColor,borderWidth.

###v0.3
Use block to handle tag's click event.
```objc
//SKTagView
@property (nonatomic, copy) void (^didClickTagAtIndex)(NSUInteger index);
```

###v0.2
Add new methods:
```objc
- (void)insertTag:(SKTag *)tag atIndex:(NSUInteger)index;
- (void)removeTag:(SKTag *)tag;
- (void)removeTagAtIndex:(NSUInteger)index;
- (void)removeAllTags;
```


Classes/ViewController.m
//
//  ViewController.m
//  WrapViewWithAutolayout
//
//  Created by shiweifu on 12/9/14.
//  Copyright (c) 2014 shiweifu. All rights reserved.
//

#import "ViewController.h"
#import "MBTagView.h"
#import <Masonry/Masonry.h>
#import <HexColors/HexColor.h>

@interface ViewController ()
@property (strong, nonatomic) MBTagView *tagView;
@property (nonatomic, strong) NSArray *colorPool;

@property (weak, nonatomic) IBOutlet UITextField *index;
@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.colorPool = @[@"#7ecef4", @"#84ccc9", @"#88abda",@"#7dc1dd",@"#b6b8de", ];
    
    [self setupTagView];
}

- (void)setupTagView
{
    self.tagView = ({
        MBTagView *view = [MBTagView new];
        view.backgroundColor = [UIColor colorWithHexString:@"#EAEAEA"];
        view.padding    = UIEdgeInsetsMake(10, 10, 10, 10);
        view.insets    = 10;
        view.lineSpace = 10;
        __weak MBTagView *weakView = view;
        weakView.didClickTagAtIndex = ^(NSUInteger index)  {
            
        };
        weakView.animateBlockForEachView = ^(UIView *v) {
            CGAffineTransform transform = v.transform;
            [UIView animateWithDuration:0.1f delay:0.0f options: UIViewAnimationOptionAutoreverse| UIViewAnimationOptionCurveLinear animations:^{
                v.transform = CGAffineTransformScale(transform, 0.95, 0.95);
            } completion:^(BOOL finished) {
                if (finished) {
                    v.transform = CGAffineTransformScale(transform, 1, 1);
                }
            }];
        };
        view;
    });
    [self.view addSubview:self.tagView];
    
    [self.tagView mas_makeConstraints:^(MASConstraintMaker *make) {
        UIView *superView = self.view;
        make.top.equalTo(superView).offset(40);
        make.leading.equalTo(superView.mas_leading);
        make.trailing.equalTo(superView.mas_trailing);
    }];
    
    [@[@"Sent", @"Attachments", @"Today", @"Conversations", @"Archieve",@"Trash", @"Others", @"Newsletters"] enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop)
     {
         MBTag *tag = [MBTag tagWithText:obj];
         tag.textColor = [UIColor whiteColor];
         tag.bgColor = [UIColor colorWithHexString:self.colorPool[arc4random() % self.colorPool.count]];;
         tag.selectedbackgroundColor = [UIColor colorWithHexString:@"#A0E07C"];
         tag.selectable = YES;
         tag.cornerRadius = 13;
         tag.animable = YES;
         tag.fontSize = 15;
         tag.padding = UIEdgeInsetsMake(5, 12, 5, 12);
         
         [self.tagView addTag:tag];
     }];
}



#pragma mark - User interactions
- (IBAction)onAdd:(id)sender
{
    MBTag *tag = [MBTag tagWithText:@"New Filter tag"];
    tag.textColor = [UIColor whiteColor];
    tag.bgColor = [UIColor colorWithHexString:self.colorPool[arc4random() % self.colorPool.count]];;
    tag.selectedbackgroundColor = [UIColor colorWithHexString:@"#A0E07C"];
    tag.selectable = YES;
    tag.cornerRadius = 13;
    tag.animable = YES;
    tag.fontSize = 15;
    tag.padding = UIEdgeInsetsMake(5, 12, 5, 12);
    
    [self.tagView addTag:tag];
}

- (IBAction)onInsert:(id)sender
{
    MBTag *tag = [MBTag tagWithText:[NSString stringWithFormat:@"Insert Tag(%ld)",(long)self.index.text.integerValue]];
    tag.textColor = [UIColor whiteColor];
    tag.bgColor = [UIColor colorWithHexString:self.colorPool[arc4random() % self.colorPool.count]];;
    tag.selectedbackgroundColor = [UIColor colorWithHexString:@"#A0E07C"];
    tag.selectable = YES;
    tag.cornerRadius = 13;
    tag.animable = YES;
    tag.fontSize = 15;
    tag.padding = UIEdgeInsetsMake(5, 12, 5, 12);
    
    [self.tagView insertTag:tag atIndex:self.index.text.integerValue];
}

- (IBAction)onRemove:(id)sender
{
    [self.tagView removeTagAtIndex:self.index.text.integerValue];
}

- (IBAction)onRemoveAll:(id)sender
{
    [self.tagView removeAllTags];
}

- (IBAction)onTapBg:(id)sender
{
    [self.view endEditing:YES];
}

@end
